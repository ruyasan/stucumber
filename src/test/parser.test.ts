import { parse } from "../lib/parser";
import { readFileSync } from "fs";

describe("Parser", () => {
  it("should parse a standard scenario", () => {
    const result = parse(readFileSync(__dirname + '/scenario.feature', 'utf8'));
    expect(result).toMatchSnapshot();
  });

  it("should parse a scenario outline", () => {
    const result = parse(readFileSync(__dirname + '/scenario-outline.feature', 'utf8'));
    expect(result).toMatchSnapshot();
  });

  it("should parse a scenario with annotations", () => {
    const result = parse(readFileSync(__dirname + '/annotations.feature', 'utf8'));
    expect(result).toMatchSnapshot();
  });

  it("should parse a scenario with data tables", () => {
    const result = parse(readFileSync(__dirname + '/data-table.feature', 'utf8'));
    expect(result).toMatchSnapshot();
  });
  it("should parse a feature with docstring", () => {
    const result = parse(readFileSync(__dirname + '/doc-string.feature', 'utf8'));
    expect(result).toMatchSnapshot();
  });

  it("should parse a feature with a background", () => {
    const result = parse(readFileSync(__dirname + '/background.feature', 'utf8'));
    expect(result).toMatchSnapshot();
  });

  it("should parse a feature with no newline at the end of the file", () => {
    const result = parse(readFileSync(__dirname + '/no-newline.feature', 'utf8'));
    expect(result).toMatchSnapshot();
  });
});
