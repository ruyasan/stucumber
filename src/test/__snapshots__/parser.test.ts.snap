// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser should parse a feature with a background 1`] = `
Object {
  "annotations": Array [],
  "background": Array [
    Object {
      "location": Object {
        "column": 10,
        "line": 3,
        "offset": 49,
      },
      "table": Array [],
      "value": "a thing",
    },
    Object {
      "location": Object {
        "column": 9,
        "line": 4,
        "offset": 66,
      },
      "table": Array [],
      "value": "I do a thing",
    },
    Object {
      "location": Object {
        "column": 9,
        "line": 5,
        "offset": 88,
      },
      "table": Array [],
      "value": "something happens",
    },
  ],
  "name": Object {
    "location": Object {
      "column": 9,
      "line": 1,
      "offset": 8,
    },
    "value": "background steps",
  },
  "scenarios": Array [
    Object {
      "annotations": Array [],
      "name": Object {
        "location": Object {
          "column": 12,
          "line": 7,
          "offset": 119,
        },
        "value": "this is a test",
      },
      "rules": Array [
        Object {
          "location": Object {
            "column": 10,
            "line": 8,
            "offset": 144,
          },
          "table": Array [],
          "value": "something",
        },
      ],
    },
  ],
}
`;

exports[`Parser should parse a feature with docstring 1`] = `
Object {
  "annotations": Array [],
  "background": null,
  "name": Object {
    "location": Object {
      "column": 9,
      "line": 1,
      "offset": 8,
    },
    "value": "doc strings",
  },
  "scenarios": Array [
    Object {
      "annotations": Array [],
      "name": Object {
        "location": Object {
          "column": 12,
          "line": 2,
          "offset": 32,
        },
        "value": "doc strings",
      },
      "rules": Array [
        Object {
          "docstring": "I have a line.
And another line.",
          "location": Object {
            "column": 10,
            "line": 3,
            "offset": 54,
          },
          "value": "I have a docstring",
        },
      ],
    },
  ],
}
`;

exports[`Parser should parse a feature with no newline at the end of the file 1`] = `
Object {
  "annotations": Array [],
  "background": null,
  "name": Object {
    "location": Object {
      "column": 9,
      "line": 1,
      "offset": 8,
    },
    "value": "background steps",
  },
  "scenarios": Array [
    Object {
      "annotations": Array [],
      "name": Object {
        "location": Object {
          "column": 12,
          "line": 3,
          "offset": 38,
        },
        "value": "this is a test",
      },
      "rules": Array [
        Object {
          "location": Object {
            "column": 10,
            "line": 4,
            "offset": 63,
          },
          "table": Array [],
          "value": "something",
        },
      ],
    },
  ],
}
`;

exports[`Parser should parse a scenario outline 1`] = `
Object {
  "annotations": Array [],
  "background": null,
  "name": Object {
    "location": Object {
      "column": 9,
      "line": 1,
      "offset": 8,
    },
    "value": "foo",
  },
  "scenarios": Array [
    Object {
      "annotations": Array [],
      "name": Object {
        "location": Object {
          "column": 20,
          "line": 2,
          "offset": 32,
        },
        "value": "this is a scenario outline",
      },
      "rules": Array [
        Object {
          "location": Object {
            "column": 10,
            "line": 3,
            "offset": 69,
          },
          "value": "that I have abc",
        },
        Object {
          "location": Object {
            "column": 9,
            "line": 4,
            "offset": 98,
          },
          "value": "I 1 2",
        },
        Object {
          "location": Object {
            "column": 9,
            "line": 5,
            "offset": 116,
          },
          "value": "I -Â£*%",
        },
      ],
    },
  ],
}
`;

exports[`Parser should parse a scenario with annotations 1`] = `
Object {
  "annotations": Array [
    Object {
      "name": "annotation1",
    },
  ],
  "background": null,
  "name": Object {
    "location": Object {
      "column": 9,
      "line": 2,
      "offset": 21,
    },
    "value": "foo",
  },
  "scenarios": Array [
    Object {
      "annotations": Array [
        Object {
          "name": "annotation2",
        },
        Object {
          "arguments": Array [
            "foo",
            1,
            Array [
              true,
              false,
            ],
          ],
          "name": "annotationWithArgs",
        },
      ],
      "name": Object {
        "location": Object {
          "column": 12,
          "line": 5,
          "offset": 99,
        },
        "value": "this is a test",
      },
      "rules": Array [
        Object {
          "location": Object {
            "column": 6,
            "line": 6,
            "offset": 120,
          },
          "table": Array [],
          "value": "a thing",
        },
      ],
    },
  ],
}
`;

exports[`Parser should parse a scenario with data tables 1`] = `
Object {
  "annotations": Array [],
  "background": null,
  "name": Object {
    "location": Object {
      "column": 9,
      "line": 1,
      "offset": 8,
    },
    "value": "data tables",
  },
  "scenarios": Array [
    Object {
      "annotations": Array [],
      "name": Object {
        "location": Object {
          "column": 12,
          "line": 2,
          "offset": 32,
        },
        "value": "data table",
      },
      "rules": Array [
        Object {
          "location": Object {
            "column": 10,
            "line": 3,
            "offset": 53,
          },
          "table": Array [
            Array [
              "datum1",
              "datum2",
            ],
            Array [
              "datum3",
              "datum4",
            ],
            Array [
              " 12 ",
              "",
            ],
          ],
          "value": "I have data",
        },
      ],
    },
  ],
}
`;

exports[`Parser should parse a standard scenario 1`] = `
Object {
  "annotations": Array [],
  "background": null,
  "name": Object {
    "location": Object {
      "column": 9,
      "line": 2,
      "offset": 30,
    },
    "value": "foo",
  },
  "scenarios": Array [
    Object {
      "annotations": Array [],
      "name": Object {
        "location": Object {
          "column": 12,
          "line": 8,
          "offset": 117,
        },
        "value": "this is a test",
      },
      "rules": Array [
        Object {
          "location": Object {
            "column": 10,
            "line": 9,
            "offset": 142,
          },
          "table": Array [],
          "value": "that I am testing",
        },
        Object {
          "location": Object {
            "column": 8,
            "line": 10,
            "offset": 168,
          },
          "table": Array [],
          "value": "I am still testing",
        },
        Object {
          "location": Object {
            "column": 9,
            "line": 11,
            "offset": 196,
          },
          "table": Array [],
          "value": "I test",
        },
        Object {
          "location": Object {
            "column": 8,
            "line": 12,
            "offset": 211,
          },
          "table": Array [],
          "value": "continue testing",
        },
        Object {
          "location": Object {
            "column": 9,
            "line": 13,
            "offset": 237,
          },
          "table": Array [],
          "value": "I want to test",
        },
        Object {
          "location": Object {
            "column": 8,
            "line": 14,
            "offset": 260,
          },
          "table": Array [],
          "value": "test",
        },
      ],
    },
  ],
}
`;
